samples/dnn/audio_visual_speech_recognition.py

```diff
@@ -0,0 +1,265 @@
+import argparse
+
+import numpy as np
+import cv2
+
+'''
+AVSpeechRecognition
+
+    How to obtain the model required for this sample:
+    Option 1: Download the model from https://drive.google.com/file/d/1xuwk5ZQagKFoTXev27zvlSg8dAmSJoo7/view?usp=sharing
+    Option 2: Convert the model using pretrained torch model and base repo.
+        Use the code here: https://gist.github.com/spazewalker/b8ab3eabc96ffcb30218cbb6f6ea09b3
```

Comment:
It's not a good solution to change the colab notebook to gist. Why do you copypaste the code from the original repo?
I have provided the code to export the model before, what is the problem with it?

Reply:
Need to hardcode the lengths to be passed in TCN. For this, I had to overwrite the forward function in AVSpeechRecog class.

Reply:
I hardcoded lengths in export function
```python
torch.onnx.export(audio_model, (audio_input, [20]), 'lipreading_audio.onnx' , opset_version=11)
```

Reply:
Initially, I was doing the same. But the model exported by this was not behaving correctly, as in this case, 20 is treated as an input and the graph comes out a little different

Reply:
>  But the model exported by this was not behaving correctly

What do you mean? I think setting constant input isn't a problem

According to documentation:
> Any non-Tensor arguments will be hard-coded into the exported model

Reply:
You can inherit your class and override the sequence length

```python
from lipreading import model
import json
from pathlib import Path
from lipreading import utils
import torch.onnx

with open('configs/lrw_resnet18_mstcn.json') as fp:
    config = json.load(fp)

class CustomLipreading(model.Lipreading):
  def forward(self, x):
     lengths = [20]
     return model.Lipreading.forward(self, x, lengths)

tcn_options = {
    'num_layers': config['tcn_num_layers'],
    'kernel_size': config['tcn_kernel_size'],
    'dropout': config['tcn_dropout'],
    'dwpw': config['tcn_dwpw'],
    'width_mult': config['tcn_width_mult'],
}

audio_model = CustomLipreading(
    num_classes=500,
    tcn_options=tcn_options,
    backbone_type=config['backbone_type'],
    relu_type=config['relu_type'],
    width_mult=config['width_mult'],
    extract_feats=False,
    modality = 'raw_audio'
)

utils.load_model(Path('models/lrw_resnet18_mstcn_audio_adamw.pth.tar'), audio_model)
audio_model.to('cpu')
audio_model.eval()

audio_input = torch.randn(1,1,10666)
torch.onnx.export(audio_model, (audio_input), 'custom_lipreading_audio.onnx' , opset_version=11, input_names=['audio_input'])
```

Reply:
I'll change this. Should i keep a kink to gist in the sample then?

Reply:
Yes, you can keep link